src/deviceDetailsChannelTableModel.cpp

```diff
@@ -0,0 +1,717 @@
+/*
+* Copyright 2017-2018, Emilien Vallot, Christophe Calmejane and other contributors
+
+* This file is part of Hive.
+
+* Hive is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+
+* Hive is distributed in the hope that it will be usefu_state,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU Lesser General Public License for more details.
+
+* You should have received a copy of the GNU Lesser General Public License
+* along with Hive.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "deviceDetailsChannelTableModel.hpp"
+
+#include <QCoreApplication>
+#include <QLayout>
+#include <QPainter>
+
+#include <la/avdecc/avdecc.hpp>
+#include <la/avdecc/controller/avdeccController.hpp>
+
+#include "ui_deviceDetailsDialog.h"
+#include "internals/config.hpp"
+#include "settingsManager/settings.hpp"
+#include "avdecc/helper.hpp"
+#include "avdecc/controllerManager.hpp"
+#include "connectionMatrix/paintHelper.hpp"
+
+
+// **************************************************************
+// class DeviceDetailsDialogImpl
+// **************************************************************
+/**
+* @brief    Private implemenation of the table model.
+* [@author  Marius Erlen]
+* [@date    2018-09-21]
+*
+* Holds a list audio cluster nodes to display in a QTableView. Supports editing of the data.
+* The data is not directly written to the controller, but stored in a map first.
+* These changes can be gathered with the getChanges method.
+*/
+class DeviceDetailsChannelTableModelPrivate : public QObject
+{
+	Q_OBJECT
+public:
+	DeviceDetailsChannelTableModelPrivate(DeviceDetailsChannelTableModel* model);
+	virtual ~DeviceDetailsChannelTableModelPrivate();
+
+	int rowCount() const;
+	int columnCount() const;
+	QVariant data(QModelIndex const& index, int role) const;
+	bool setData(const QModelIndex& index, const QVariant& value, int role);
+	QVariant headerData(int section, Qt::Orientation orientation, int role) const;
+	Qt::ItemFlags flags(QModelIndex const& index) const;
+
+	void registerUiWidget(QTableView* ui);
```

Comment:
Seems strange to have something referring to the view in the model.

Reply:
I agree on this and we have removed that in the latest commits. It was really not needed after all.